Separate test store for shortcuts test

shresthabijay opened the pull request:
fixes #781

shresthabijay commented:
@raineorshine I have exported a factory function that takes `store` as param and encapsulates shortcut handlers. I have made necessary changes to the places where these handlers were used. I also created a `createTestStore` function that helps create new store for every shortcut tests. Please review the changes. Thanks!

raineorshine commented on src/shortcuts.js:
```diff
@@ -84,100 +83,106 @@ const isGestureHint = state => state.alert && state.alert.alertType === 'gesture
-  setTimeout(() => {
-    if (isGestureHint(store.getState())) {
-      alert(null)
+/**
+ * Keyboard handlers factory function.
+ */
+export const keyboardHandlers = store => ({
+
+  /** Handles gesture hints when a valid segment is entered. */
+  handleGestureSegment: (g, sequence, e) => {
+
```
Since `handleGestureSegment` and `handleGestureEnd` are not keyboard inputs, a better name would be `inputHandlers`.

shresthabijay replied:
Sure!

raineorshine commented on src/shortcuts/__tests__/newThoughtOrOutdent.js:
```diff
@@ -1,10 +1,12 @@
 import { initialize } from '../../initialize'
+import { createTestStore } from '../../test-helpers/createTestStore'
+
+const store = createTestStore()
```
This should probably be assigned in `beforeEach`, otherwise data could carry over between tests.

shresthabijay commented:
@raineorshine I have removed `initialize` from the shortcut tests and create new store within the scope of each test. I renamed `keyboardHandlers` to `inputHandlers`. Please review the changes. Thanks!
